### DISCUSSION TITLE
Problems recreating "Animating A Re-Ordered List"

### DISCUSSION THREAD

**Discussion by sERvAss** [livewire/livewire#5120] [1. Help] [answered]
Hello,

I'm trying to follow the screencast called "Animating A Re-Ordered List".
https://laravel-livewire.com/screencasts/s8-animate-move
It is an example of how you can animate elements that are changing positions using javascript/livewire.

I can easily replicate the first example where only the top element of the list moves.
Everything works fine there and the top element moves to its new location smoothly every time.
But once I change the code to an array for all the elements the animations are not triggered anymore.

My code is identical to the screencast:

```
<script type="text/javascript">
    document.addEventListener("DOMContentLoaded", () => {
        let animations = []

        Livewire.hook('message.received', () => {
            let things = Array.from(document.querySelectorAll('[animate-move]'))

            console.log('Hook 1: ok')

            animations = things.map(thing => {
                let oldTop = thing.getBoundingClientRect().top

                console.log('Map Function: ok')

                return () => {
                    let newTop = thing.getBoundingClientRect().top

                    console.log('Callback: ok')

                    thing.animate([{
                            transform: `translateY(${oldTop - newTop}px)`
                        },
                        {
                            transform: `translateY(0px)`
                        },
                    ], {
                        duration: 1000,
                        easing: 'ease'
                    })
                }
            })
        });

        Livewire.hook('message.processed', () => {
            while (animations.lenght) {
                animations.shift()()
            }

            console.log('Hook 2: ok')
        });
    });
</script>
```

The only things I added are "debug" logs to see if all parts of the code are excecuted.
The problem seems to be the callback. It does not get triggered.

The rest of the code (Blade Component, Livewire PHP File) is identical to the demo screencast and works fine.
Elements of the list are loaded and shuffled on click.
There are no errors shown in console.

![Screenshot 2022-07-22 145013](https://user-images.githubusercontent.com/93264186/180443147-220bb99c-eb84-4427-b99e-0ec65d4247e7.png)

I have never used a return function in that way before, so I'm having trouble diagnosing the error.
Can someone please help me or point me in the right direction?

Thank you for your time.

**Comment by joshhanley** [comment]
@sERvAss have you tried animating them just using CSS instead of using Javascript?

**Comment by Martleby** [comment, marked as answer]
It's not something as simple as a typo is it?

```
while (animations.lenght) {
                animations.shift()()
            }
```
You've got **lenght** instead of "length" ...

**Reply by joshhanley** [reply]
Good catch! You might be right! 🙂

**Reply by sERvAss** [reply]
OMG. I'm such a doofus.
Thank you very much @Martleby !!

Interestingly neither VSCode nor the Console showed this as an error.
But now it works exactly as shown in the screencast.
And now I can implement it into my own project.
Thanks again!

**Reply by Martleby** [reply]
Ha, glad it worked for you! Don't be too hard on yourself, though... I've been chasing these sorts of bugs in my own code since I started learning Basic on my Beeb back in the 1980's 😁
